python/tank/bootstrap/manager.py

```diff
@@ -546,37 +560,30 @@ def get_pipeline_configurations(self, project):
             If ``None``, this will enumerate the pipeline configurations
             for the site configuration.
         :type project: Dictionary with keys ``type`` and ``id``.
+        :param list pc_entities: A list of PipelineConfiguration entity dictionaries. This
+            can be used to pass in pre-queried entities to take advantage of the filtering
+            and ordering functionality without the need to re-query the data from Shotgun.
 
         :returns: List of pipeline configurations.
         :rtype: List of dictionaries with keys ``type``, ``id``, ``name`` and ``project``. The pipeline
             configurations will always be sorted such as the primary pipeline configuration, if available,
             will be first. Then the remaining pipeline configurations will be sorted by ``name`` field
             (case insensitive), then the ``project`` field and finally then ``id`` field.
         """
-
-        if isinstance(self.pipeline_configuration, int):
+        if pc_entities is None and isinstance(self.pipeline_configuration, int):
             raise TankBootstrapError("Can't enumerate pipeline configurations matching a specific id.")
 
         resolver = ConfigurationResolver(
             self.plugin_id,
             project["id"] if project else None
         )
 
-        # Only return id, type and code fields.
-        pcs = []
-        for pc in resolver.find_matching_pipeline_configurations(
+        return resolver.find_matching_pipeline_configurations(
```

Comment:
What I wrote above is a lie. I now have no recollection as to why I made this specific change. Reverting it now.